### PR TITLE
Primary  attributes

### PR DESCRIPTION
**Pull request by fogelito**
Previous PR
https://github.com/utopia-php/database/pull/243

## Summary by CodeRabbit

* **New Features**
  * Introduced support for a new `id` attribute type with explicit validation for integer and string (MongoDB ObjectId) formats.
  * Enabled collections and documents to include and index the new `id` attribute type.
  * Added sequence value validation tied to the `id` attribute type.
  * Database adapters now recognize and handle the new `id` attribute type with appropriate SQL type mappings.

* **Bug Fixes**
  * Improved handling of tenant and sequence fields for consistent data typing and validation.

* **Tests**
  * Extended test coverage for the `id` attribute type, including creation, querying, edge cases (null, zero), and validation behavior across different ID types.
  * Updated validators and query tests to incorporate the new `id` attribute type.

* **Documentation**
  * Clarified attribute type definitions and validation logic through updated in-code comments.

### CODE REVIEW

File: src/Database/Validator/Sequence.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        switch ($this->idAttributeType) {
+            case Database::VAR_ID_MONGO:
+                return preg_match('/^[a-f0-9]{24}$/i', $value) === 1;
+
+            case Database::VAR_ID_INT:
+                $validator = new Integer(loose: true);
+                if (!$validator->isValid($value)) {
+                    return false;
+                }
+
```
Comment (abnegate) [inline src/Database/Validator/Sequence.php:58]:
```suggestion

```

**Review by abnegate — APPROVED** [review]
